superset-frontend/src/utils/simpleFilterToAdhoc.ts

```diff
@@ -18,20 +18,27 @@
  */
 import {
   AdhocFilter,
+  createTimeRangeFromGranularity,
   isAdhocColumn,
   QueryObjectFilterClause,
   SimpleAdhocFilter,
+  TimeGranularity,
 } from '@superset-ui/core';
 import {
   CLAUSES,
   EXPRESSION_TYPES,
 } from '../explore/components/controls/FilterControl/types';
-import { OPERATOR_ENUM_TO_OPERATOR_TYPE } from '../explore/constants';
+import {
+  OPERATOR_ENUM_TO_OPERATOR_TYPE,
+  Operators,
+} from '../explore/constants';
 import { translateToSql } from '../explore/components/controls/FilterControl/utils/translateToSQL';
 
 export const simpleFilterToAdhoc = (
   filterClause: QueryObjectFilterClause,
   clause: CLAUSES = CLAUSES.WHERE,
+  isTemporal = false,
+  timeGrain: TimeGranularity = 'P1D',
```

Comment:
I don't think we should assume any time grain here - let's just leave it optional